metrics/web/test/common/presentation/navigation/state/navigation_notifier_test.dart

```diff
@@ -168,18 +169,149 @@ void main() {
+
+        expect(notifier.pages, hasLength(2));
+      },
+    );
+
+    test(
+      ".handleLoggedOut() does not push to the login page when a is not logged in",
+      () {
+        final initialPagesLength = notifier.pages.length;
+
+        notifier.handleLoggedOut();
+
+        final actualPagesLength = notifier.pages.length;
+
+        expect(actualPagesLength, equals(initialPagesLength));
+      },
+    );
+
+    test(
+      ".handleLoggedIn() pushes the dashboard route if the current redirect route is null",
+      () {
+        notifier.handleLoggedIn();
+
+        final currentConfigurationName = notifier.currentConfiguration.name;
+
+        expect(currentConfigurationName, equals(RouteName.dashboard));
+      },
+    );
+
+    test(
+      ".handleLoggedIn() pushes the loading route if the current redirect route is loading",
+      () {
+        notifier.handleAppInitialized(isAppInitialized: false);
+        notifier.push(DefaultRoutes.loading);
+
+        notifier.handleLoggedIn();
+
+        final currentConfigurationName = notifier.currentConfiguration.name;
+
+        expect(currentConfigurationName, equals(RouteName.loading));
+      },
+    );
+
+    test(
+      ".handleLoggedIn() clears pages before redirect",
+      () {
+        notifier.handleAppInitialized(isAppInitialized: true);
+
+        notifier.handleLoggedIn();
+
+        expect(notifier.pages, hasLength(equals(1)));
+      },
+    );
+
+    test(
+      ".handleLoggedIn() pushes the redirect route if it is not null",
+      () {
+        final configuration = DefaultRoutes.projectGroups;
+
+        notifier.handleAppInitialized(isAppInitialized: false);
+        notifier.handleInitialRoutePath(configuration);
+        notifier.handleAppInitialized(isAppInitialized: true);
+
+        notifier.handleLoggedIn();
+
+        final currentConfigurationName = notifier.currentConfiguration.name;
+
+        expect(currentConfigurationName, configuration.name);
+      },
+    );
+
+    test(
+      ".handleLoggedIn() replaces the current navigation state using the redirect route path",
+      () {
+        final configuration = DefaultRoutes.projectGroups;
+        notifier.push(configuration);
+
+        notifier.handleLoggedIn();
+
+        verify(
+          navigationState.replaceState(any, any, configuration.path),
+        ).called(once);
+      },
+    );
+
+    test(
+      ".handleLoggedIn() clears the redirect route after redirect if user is logged in",
+      () {
+        notifier.handleInitialRoutePath(DefaultRoutes.projectGroups);
+
+        notifier.handleLoggedIn();
+
+        notifier.handleAppInitialized(isAppInitialized: true);
+
+        final currentConfigurationName = notifier.currentConfiguration.name;
+
+        expect(currentConfigurationName, equals(RouteName.dashboard));
+      },
+    );
+
+    test(
+      ".handleLoggedIn() clears the redirect route after redirect if the route does not require authorization",
```

Comment:
✅ 
It seems like this method will clear the redirect route in any case, so I think this test is redundant. Also, should we test that this method does nothing if the user is already logged in?